File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int maxi = 0;
        int mini = 0;
        
        for(int i=1;i<prices.size();i++){
            int profit = prices[i]- prices[mini];
            maxi = max(maxi, profit);
            if(prices[i] < prices[mini]) mini = i;
        }
        return maxi;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        
        mini = prices[0]
        maxi = 0

        for i in range(len(prices)):
            if mini > prices[i]:
                mini = prices[i]
            profit = prices[i]-mini
            maxi = max(maxi, profit)
        
        return maxi
        